RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducer.cc

```diff
@@ -0,0 +1,83 @@
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitHostCollection.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+#include "HeterogeneousCore/CUDACore/interface/JobConfigurationGPURecord.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/alpaka/PFClusterParamsDeviceCollection.h"
+#include "RecoParticleFlow/PFClusterProducer/plugins/alpaka/PFClusterSoAProducerKernel.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+
+#define DEBUG false
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+  class PFClusterSoAProducer : public stream::EDProducer<> {
+  public:
+    PFClusterSoAProducer(edm::ParameterSet const& config)
+        : pfClusParamsToken(esConsumes(config.getParameter<edm::ESInputTag>("pfClusterParams"))),
+          inputPFRecHitSoA_Token_{consumes(config.getParameter<edm::InputTag>("PFRecHitsLabelIn"))},
+          outputPFClusterSoA_Token_{produces()},
+          outputPFRHFractionSoA_Token_{produces()},
+          synchronise_(config.getParameter<bool>("synchronise")),
+          produceSoA_{config.getParameter<bool>("produceSoA")} {}
+
+    void produce(device::Event& event, device::EventSetup const& setup) override {
+      const reco::PFClusterParamsDeviceCollection& params = setup.getData(pfClusParamsToken);
+      const reco::PFRecHitHostCollection& pfRecHits = event.get(inputPFRecHitSoA_Token_);
+      const int nRH = pfRecHits->size();
+
+      reco::PFClusteringVarsDeviceCollection pfClusteringVars{nRH + 1, event.queue()};
+      reco::PFClusteringEdgeVarsDeviceCollection pfClusteringEdgeVars{(nRH * 8) + 1, event.queue()};
+      reco::PFClusterDeviceCollection pfClusters{nRH, event.queue()};
+      reco::PFRecHitFractionDeviceCollection pfrhFractions{nRH * 120, event.queue()};
```

Comment:
what is 120 ?

Reply:
OK.
Could you make `120` a parameter that can be configured from python, and give it a name and a comment that reflects what you wrote here ?

Reply:
The rationale being the, if we later need to increase the limit, it's going to be easier if we can do it from the configuration.